specs/work_queue_spec.rb

```diff
@@ -38,6 +38,31 @@ def wait_for_task_to_complete(task: nil)
       expect(ensured).to be(true)
     end
 
+    it 'Reports success state when task completed without exceptions' do
+      queue = TaskQueue.new(name: 'test queue')
+      work_completed = false
+      success_task = false
+      task = Task.new(work_block: proc { work_completed = true }, ensure_block: proc { |success| success_task = true })
+      queue.add_task_async(task: task)
+      wait_for_task_to_complete(task: task)
+      expect(work_completed).to be(true)
+      expect(success_task).to be(true)
```

Comment:
What do you think about adding `expect(task.finished_successfully).to be(true)` to all of these to verify internal state too?